packages/frontend/src/components/StrikeWarning.tsx

```diff
@@ -89,18 +90,22 @@ export default function StrikeModalManager() {
   const isWalletConnected = useAtomValue(connectedWalletAtom)
   const [isStrikeModalShownOnce, setIsStrikeModalShownOnce] = useState(false)
   const { isRestricted } = useRestrictUser()
+  const router = useRouter()
+
+  // Check if the current page is either terms of service or privacy policy
+  const isExcludedPage = ['/terms-of-service', '/privacy-policy'].includes(router.pathname)
 
   // delayed state change - show modal after 2 seconds
   // this is because it takes some time to automatically connect the wallet (incase previously connected)
   useEffect(() => {
-    if (isRestricted) {
+    if (isRestricted && !isExcludedPage) {
```

Comment:
i don't think we need to check `isExcludedPage` here. 

`if (showModal && !isExcludedPage)` this should be enough

Reply:
Awesome catch, you're right this is redundant. Fixed in new version.